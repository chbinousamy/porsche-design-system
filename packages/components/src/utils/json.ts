export const parseJSONAttribute = <T>(attribute: T | string): T => {
  // Input is object, e.g. { 'aria-label': 'Some label' }
  if (typeof attribute !== 'string') {
    return attribute;
  }

  // Convert single quotes to double quotes except the ones which are escaped by backslash
  let jsonString = attribute
    .replace(/\\'/g, '__escaped_single_quote__')
    .replace(/'/g, '"')
    .replace(/__escaped_single_quote__/g, '\\\'');

  // Remove string escapes except the ones followed by unicode u0027
  jsonString = jsonString.replace(/([^\\])\\(?!u0027)/g, '$1');

  // Wrap keys in double quotes
  jsonString = jsonString.replace(/[\s"]?([\w-]+)[\s"]?:/g, '"$1":');

  return JSON.parse(jsonString);
};
