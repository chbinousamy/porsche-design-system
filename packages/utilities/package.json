{
  "name": "utilities-v2",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "build": "yarn build:utilities && yarn build:ag-grid",
    "build:utilities": "yarn --cwd ./projects/utilities build",
    "build:ag-grid": "yarn --cwd ./projects/ag-grid build",
    "test": "yarn test:unit",
    "test:unit": "yarn --cwd ./projects/utilities test:unit",
    "clean:build": "yarn --cwd ./projects/utilities clean:build"
  },
  "volta": {
    "extends": "../../package.json"
  }
}
