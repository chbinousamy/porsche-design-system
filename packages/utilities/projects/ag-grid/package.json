{
  "name": "@porsche-design-system/ag-grid",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "sideEffects": false,
  "scripts": {
    "build": "yarn clean:build && yarn build:sass && yarn build:bundle",
    "build:sass": "sass src/scss:src/css --load-path=../../../../node_modules",
    "build:bundle": "rollup -c rollup.config.js --bundleConfigAsCjs",
    "test": "yarn test:unit",
    "test:unit": "vitest --config tests/unit/config/vitest.config.ts",
    "clean:build": "rm -rf ./dist && rm -rf ./src/css"
  },
  "dependencies": {
    "@porsche-design-system/components-js": "3.18.0"
  },
  "peerDependencies": {
    "sass": "*"
  },
  "volta": {
    "extends": "../../package.json"
  }
}
