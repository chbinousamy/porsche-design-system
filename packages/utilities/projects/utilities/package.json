{
  "name": "@porsche-design-system/utilities-v2",
  "version": "0.0.0",
  "private": true,
  "license": "UNLICENSED",
  "scope": "@porsche-design-system",
  "main": "dist/js/index.js",
  "module": "dist/js/esm/index.js",
  "types": "dist/js/index.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "yarn clean:build && yarn build:generateScss && yarn build:bundle",
    "build:generateScss": "ts-node scripts/generateScss.ts",
    "build:bundle": "rollup -c rollup.config.js --bundleConfigAsCjs",
    "test": "yarn test:unit",
    "test:unit": "NODE_OPTIONS=--experimental-vm-modules jest --colors -c tests/unit/config/jest.config.js",
    "clean": "yarn clean:node_modules && yarn clean:build",
    "clean:node_modules": "rm -rf ./node_modules && rm -rf ./yarn-error.log",
    "clean:build": "rm -rf ./dist && rm -rf ./src/scss/lib"
  },
  "dependencies": {},
  "devDependencies": {
    "@porsche-design-system/shared": "0.0.0",
    "@types/madge": "^5.0.0",
    "@types/styled-components": "^5.1.26",
    "change-case": "^4.1.2",
    "jest": "^29.6.2",
    "madge": "^6.1.0",
    "prettier": "^3.0.1",
    "react-jss": "^10.10.0",
    "sass": "^1.64.2",
    "styled-components": "^5.3.11",
    "ts-jest": "^29.1.1",
    "ts-node": "^10.9.1",
    "typescript": "~5.1.6"
  },
  "volta": {
    "extends": "../../package.json"
  }
}
