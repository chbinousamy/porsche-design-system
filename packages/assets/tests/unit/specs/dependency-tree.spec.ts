// @ts-ignore
import madge from 'madge';
import * as path from 'path';
import { main, module } from '../../../dist/package.json';

it('should have no regression for dependency tree', async () => {
  const packageDir = path.resolve(__dirname, '../../../dist');
  const treeCjs = await madge(`${packageDir}/${main}`);
  const treeEsm = await madge(`${packageDir}/${module}`);

  // visualize result
  // await tree.image('result.svg')

  expect(treeCjs.obj()).toMatchSnapshot();
  expect(treeEsm.obj()).toMatchSnapshot();
});
