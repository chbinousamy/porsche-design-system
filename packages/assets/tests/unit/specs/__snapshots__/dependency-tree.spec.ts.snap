// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have no regression for dependency tree 1`] = `
{
  "index.js": [],
}
`;

exports[`should have no regression for dependency tree 2`] = `
{
  "index.js": [],
}
`;
